packages/mui-material/src/usePagination/usePagination.js

```diff
@@ -63,7 +63,7 @@ export default function usePagination(props = {}) {
       boundaryCount + siblingCount * 2 + 2,
     ),
     // Less than endPages
-    endPages.length > 0 ? endPages[0] - 2 : count - 1,
+    count - boundaryCount - 1,
```

Comment:
This new approach works, but what's the advantage over the previous one? The earlier method might handle edge cases better and is more readable to me, as `endPages[0] - 2` clearly indicates that the sibling pages  end (`siblingsEnd`) before the start of the end pages.

Reply:
The new approach offers advantages such as less code and fewer conditional statements, but the benefits are minimal.

I absolutely agree with your opinion, but readability is quite subjective. 

When I first tried to understand how `siblingsEnd` was calculated, I did not grasp why there was a need to check `endPages.length > 0`. I had to first look at what `endPages` is and figure out when `endPages` would be empty.

I think the original version of `siblingsEnd` would not be too difficult for those familiar with the entire code, but the new version should be easier for newcomers to understand.

Reply:
Alright, makes sense to me.